Model/SentryInteraction.php

```diff
@@ -21,14 +21,17 @@
 
 class SentryInteraction
 {
+    /**
+     * @var ?UserContextInterface
+     */
+    private ?UserContextInterface $userContext = null;
+
     /**
      * SentryInteraction constructor.
      *
-     * @param UserContextInterface $userContext
-     * @param State                $appState
+     * @param State $appState
      */
     public function __construct(
-        private UserContextInterface $userContext,
```

Comment:
I guess you could also try to use `Magento\Authorization\Model\UserContextInterfaceFactory` here to initialize the userContextInterface on demand later on. Using the Factory class should prevent automatically initializing of the class.